WooCommerce/Classes/Tools/UnitInputFormatter/ShippingInputFormatter.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// `UnitInputFormatter` implementation for shipping input.
+///
+struct ShippingInputFormatter: UnitInputFormatter {
+
+    // Match all the characters not included in the set: numbers between 0 and 9, and - . , %
+    let regex = "[^-0-9%.,]"
+
+    func isValid(input: String) -> Bool {
+        guard input.isEmpty == false else {
+            // Allows empty input to be replaced by 0.
+            return true
+        }
+
+        return input.range(of: regex, options: .regularExpression) == nil
+    }
+
+    func format(input text: String?) -> String {
+        guard let text = text, text.isEmpty == false else {
+            return "0"
```

Comment:
yea when I made the `DecimalInputFormatter`, it defaults to `"0"` with my own assumption from the design. since it's possible to unset the value of a weight/dimension field in wp-admin and could be actual use cases, let's return an empty string here?

Reply:
unrelated to this PR, we might want to do the same for the price fields, allowing empty string!

Reply:
I agree to return an empty string here. 
I also opened a separated issue to handle zero values on shipping settings https://github.com/woocommerce/woocommerce-ios/issues/1785 and price settings https://github.com/woocommerce/woocommerce-ios/issues/1786